ydb/core/grpc_services/grpc_request_proxy.cpp

```diff
@@ -135,34 +136,32 @@ class TGRpcRequestProxyImpl
         return true;
     }
 
-    template <typename TEvent>
-    void PreHandle(TAutoPtr<TEventHandle<TEvent>>& event, const TActorContext& ctx) {
+    template<class TEvent>
+    bool PreHandleImpl(TAutoPtr<TEventHandle<TEvent>>& event, const TActorContext& ctx) {
         IRequestProxyCtx* requestBaseCtx = event->Get();
-
-        LogRequest(event);
-
         if (!SchemeCache) {
             const TString error = "Grpc proxy is not ready to accept request, no proxy service";
             LOG_ERROR_S(ctx, NKikimrServices::GRPC_SERVER, error);
             const auto issue = MakeIssue(NKikimrIssues::TIssuesIds::GENERIC_TXPROXY_ERROR, error);
             requestBaseCtx->RaiseIssue(issue);
             requestBaseCtx->ReplyWithYdbStatus(Ydb::StatusIds::UNAVAILABLE);
-            return;
+            return true;
         }
 
-
-        MaybeStartTracing(*requestBaseCtx);
+        if (!IsHandlingDeferred) {
+            MaybeStartTracing(*requestBaseCtx);
+        }
 
         if (IsAuthStateOK(*requestBaseCtx)) {
             Handle(event, ctx);
-            return;
+            return false;
```

Comment:
There is an issue here. If Auth is not required span will not be finished.
I think possible solution in this case -  just skip this first span